xbmc/peripherals/bus/PeripheralBus.h

```diff
@@ -54,7 +54,7 @@ namespace PERIPHERALS
     /*!
      * @return True if this bus needs to be polled for changes, false if this bus performs updates via callbacks
      */
-    bool NeedsPolling(void) const { return m_bNeedsPolling; }
+    bool NeedsPolling(void) const { CSingleLock lock(m_critSection); return m_bNeedsPolling; }
```

Comment:
I'm was talking about the high level ones, such as std::atomic::compare_exchange_strong which could replace locks in the scenario, sorry if that was confusing..

But this is off topic, so decision time: this or https://github.com/xbmc/xbmc/compare/master...tamland:fix_peripherals_deadlock_alt ?